src/diffusers/pipelines/z_image/pipeline_z_image.py

```diff
@@ -417,8 +407,8 @@ def __call__(
                 f"Please adjust the width to a multiple of {vae_scale}."
             )
 
-        assert self.dtype == torch.bfloat16
-        dtype = self.dtype
+        # assert self.dtype == torch.bfloat16
+        dtype = self.dtype if hasattr(self, "dtype") and self.dtype is not None else torch.float32
```

Comment:
we usually don't use `self.dtype` , this is the logic behind it https://github.com/huggingface/diffusers/blob/main/src/diffusers/pipelines/pipeline_utils.py#L578

you can see that it is not very useful when components can sometimes have different dtype
so instead, we try to use specific dtype at each step, e.g. you will see a lot of patterns like this 
https://github.com/huggingface/diffusers/blob/main/src/diffusers/pipelines/flux/pipeline_flux.py#L257

```py
prompt_embeds = encode_prompt(..., dtype= self.text_encoder.dtype)
```

or 

``` py
latents = prepare_latents(...,  dtype=torch.float32)
```